From e5d6140c2d89ee45d18d0814984a1e778b72338b Mon Sep 17 00:00:00 2001
From: yzha108 <yanli.l.zhang@intel.com>
Date: Mon, 9 Jul 2018 18:52:49 +0800
Subject: [PATCH 2003/2367] media: intel-ipu4: ar0231: ioctl parameters issue.

Message for Open Source:
Make sure IOCTL parameters value is correct

Message for Internal:
[Issue/Feature]
Make sure IOCTL parameters value is correct

[Root Cause/Changes]
Set AG/DG/Exposure time min/max value correctly

Change-Id: I6f92f31e2c0ade777ce57faa5d1bb3e84aac5284
TrackedOn: #JIIAP-531
Signed-off-by: yzha108 <yanli.l.zhang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 .../i2c/crlmodule/crl_ar0231at_configuration.h     | 34 +++++++++++-----------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h b/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
index 697e35e..fc639c0 100644
--- a/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
+++ b/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
@@ -1748,10 +1748,10 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = V4L2_CID_ANALOGUE_GAIN,
 		.name = "V4L2_CID_ANALOGUE_GAIN",
 		.type = CRL_V4L2_CTRL_TYPE_INTEGER,
-		.data.std_data.min = 0,
-		.data.std_data.max = 16,
+		.data.std_data.min = 0x0000,
+		.data.std_data.max = 0xFFFF,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0,
+		.data.std_data.def = 0xAAAA,
 		.flags = 0,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
@@ -1767,7 +1767,7 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = V4L2_CID_GAIN,
 		.name = "Digital Gain",
 		.type = CRL_V4L2_CTRL_TYPE_INTEGER,
-		.data.std_data.min = 0x0001,
+		.data.std_data.min = 0x0080,
 		.data.std_data.max = 0x07FF,
 		.data.std_data.step = 1,
 		.data.std_data.def = 0x0080,
@@ -1809,7 +1809,7 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.data.std_data.min = 0,
 		.data.std_data.max = 65535,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0x0,
+		.data.std_data.def = 0x0222,
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
@@ -1826,10 +1826,10 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = V4L2_CID_EXPOSURE,
 		.name = "T1_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_INTEGER,
-		.data.std_data.min = 1,
-		.data.std_data.max = 65535,
+		.data.std_data.min = 0x0,
+		.data.std_data.max = 0xFFFF,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0x40,
+		.data.std_data.def = 0x0163,
 		.flags = 0,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
@@ -1845,10 +1845,10 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = CRL_CID_EXPOSURE_SHS1,
 		.name = "T2_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_CUSTOM,
-		.data.std_data.min = 1,
-		.data.std_data.max = 65535,
+		.data.std_data.min = 0x0,
+		.data.std_data.max = 0xFFFF,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0x40,
+		.data.std_data.def = 0x0002,
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
@@ -1865,10 +1865,10 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = CRL_CID_EXPOSURE_SHS2,
 		.name = "T3_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_CUSTOM,
-		.data.std_data.min = 1,
-		.data.std_data.max = 65535,
+		.data.std_data.min = 0x0,
+		.data.std_data.max = 0xFFFF,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0x40,
+		.data.std_data.def = 0x0001,
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
@@ -1885,10 +1885,10 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = CRL_CID_EXPOSURE_SHS3,
 		.name = "T4_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_CUSTOM,
-		.data.std_data.min = 1,
-		.data.std_data.max = 65535,
+		.data.std_data.min = 0x0,
+		.data.std_data.max = 0xFFFF,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0x40,
+		.data.std_data.def = 0x0,
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
-- 
2.7.4

